src/App.tsx

```diff
@@ -1,11 +1,70 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserWarning } from './components/UserWarning/UserWarning';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Errors } from './components/Errors/Errors';
+import { Status } from './types/Status';
+import { ErrorSpec } from './types/ErrorSpec';
+
+const USER_ID = 12021;
+const ADDED_URL = `/todos?userId=${USER_ID}`;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [hasErrors, setHasErrors] = useState(false);
+  const [error, setError] = useState<ErrorSpec | null>(null);
+  const [status, setStatus] = useState<Status>(Status.ALL);
+  const [uncompletedTodosCount, setUncompletedTodosCount] = useState<number>(0);
+
+  useEffect(() => {
+    client
+      .get<Todo[]>(ADDED_URL)
+      .then((todos) => {
+        setTodosFromServer(todos);
+
+        if (todos.length === 0) {
+          setError(ErrorSpec.EMPTY_TITLE);
+        }
+      })
+      .catch(() => setError(ErrorSpec.NOT_LOADED));
+  }, []);
+
+  useEffect(() => {
+    const completedCount = todosFromServer.filter(
+      todo => !todo.completed,
+    ).length;
+
+    setUncompletedTodosCount(completedCount);
+  }, [todosFromServer, status]);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case Status.ACTIVE:
+        return todosFromServer.filter(todo => !todo.completed);
+
+      case Status.COMPLETED:
+        return todosFromServer.filter(todo => todo.completed);
+
+      default:
+        return todosFromServer;
+    }
```

Comment:
Filtering logic should be brought out as a separate helper outside of components to improve code readability